ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoContractsAppendOnlySpec.scala

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao
+
+import com.daml.lf.transaction.GlobalKey
+import com.daml.lf.transaction.Node.KeyWithMaintainers
+import com.daml.lf.value.Value.ValueText
+import com.daml.platform.store.interfaces.LedgerDaoContractsReader
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{LoneElement, OptionValues}
+
+private[dao] trait JdbcLedgerDaoContractsAppendOnlySpec extends LoneElement with OptionValues {
+  this: AsyncFlatSpec with Matchers with JdbcLedgerDaoSuite =>
+
+  private def contractsReader = ledgerDao.contractsReader
+
+  behavior of "JdbcLedgerDao (contracts) on append-only schema"
+
+  it should "present the contract state at a specific event sequential id" in {
+    for {
+      (_, tx) <- store(singleCreate(create(_, signatories = Set(alice))))
+      contractId = nonTransient(tx).loneElement
+      _ <- store(singleNonConsumingExercise(contractId))
+      eventSeqIdAtCreate <- ledgerDao.lookupLedgerEndOffsetAndSequentialId()
```

Comment:
```suggestion
      (_, eventSeqIdAtCreate) <- ledgerDao.lookupLedgerEndOffsetAndSequentialId()
```

This will save you having to write `eventSeqIdAtCreate._2` below, and make the variable name more correct. Same for all other calls to `lookupLedgerEndOffsetAndSequentialId`.

Reply:
Good point, done.